src/system.transactions/tests/LTMEnlistmentTests.cs

```diff
@@ -0,0 +1,192 @@
+                        TestEnlistment[] volatiles = new TestEnlistment[volatileCount];
+                        for (int i = 0; i < volatileCount; i++)
+                        {
+                            // It doesn't matter what we specify for SinglePhaseVote.
+                            volatiles[i] = new TestEnlistment(volatilePhase1Vote, expectedVolatileOutcome);
+                            tx.EnlistVolatile(volatiles[i], volatileEnlistmentOption);
+                        }
+                    }
+
+                    TestEnlistment durable = new TestEnlistment(Phase1Vote.Prepared, expectedDurableOutcome);
+                    Assert.Throws<PlatformNotSupportedException>(() => // Creation of two phase durable enlistment attempts to promote to MSDTC
+                    {
+                        tx.EnlistDurable(Guid.NewGuid(), durable, durableEnlistmentOption);
+                    });
+
+                    if (commit)
+                    {
+                        ts.Complete();
+                    }
+                }
+            }
+            catch (TransactionInDoubtException)
+            { }
+            catch (TransactionAbortedException)
+            { }
+
+            Assert.True(tx != null, "Transaction is not expected to be null");
+            Assert.True(tx.TransactionInformation.Status == expectedTxStatus, "Unexpected TransactionStatus - " + tx.TransactionInformation.Status.ToString());
+        }
+
+        [Theory]
+        [InlineData(EnlistmentOptions.EnlistDuringPrepareRequired, EnlistmentOptions.None, Phase1Vote.Prepared, true, true, EnlistmentOutcome.Committed, TransactionStatus.Committed)]
+        [InlineData(EnlistmentOptions.None, EnlistmentOptions.None, Phase1Vote.Prepared, false, true, EnlistmentOutcome.Committed, TransactionStatus.Committed)]
+        public void EnlistDuringPhase0(EnlistmentOptions enlistmentOption, EnlistmentOptions phase0EnlistmentOption, Phase1Vote phase1Vote, bool expectPhase0EnlistSuccess, bool commit, EnlistmentOutcome expectedOutcome, TransactionStatus expectedTxStatus)
+        {
+            Transaction tx = null;
+            AutoResetEvent outcomeEvent = null;
+            try
+            {
+                using (TransactionScope ts = new TransactionScope())
+                {
+                    tx = Transaction.Current.Clone();
+                    outcomeEvent = new AutoResetEvent(false);
+                    TestEnlistment enlistment = new TestEnlistment(phase1Vote, expectedOutcome, true, expectPhase0EnlistSuccess, outcomeEvent);
+                    tx.EnlistVolatile(enlistment, enlistmentOption);
+
+                    if (commit)
+                    {
+                        ts.Complete();
+                    }
+                }
+            }
+            catch (TransactionInDoubtException)
+            { }
+            catch (TransactionAbortedException)
+            { }
+
+            Assert.True(outcomeEvent.WaitOne(TimeSpan.FromSeconds(5)));
```

Comment:
Is this ```5``` just a value chosen somewhat randomly to provide a sufficiently long enough time for something to happen?  Do we expect the test to take a long time?  Consider a) putting the 5 into a named const, and b) if we expect this test to take seconds adding [OuterLoop] to it so it doesn't slow down the "inner loop".